Doc/c-api/frame.rst

```diff
@@ -0,0 +1,75 @@
+.. highlight:: c
+
+Frame Objects
+-------------
+
+.. c:type:: PyFrameObject
+
+   The C structure of the objects used to describe frame objects.
+
+   The structure is only part of the internal C API: fields should not be
+   access directly. Use getter functions like :c:func:`PyFrame_GetCode` and
+   :c:func:`PyFrame_GetBack`.
+
+   Debuggers and profilers can use the internal C API to access this structure
+   without calling functions, but the internal C API doesn't provide any
+   backward compatibility warranty.
+
+   .. versionchanged:: 3.11
+      The structure moved to the internal C API headers.
+
+Public members:
```

Comment:
It would be much safer to just say "don't access fields directly".

For example, `f_lineno` is not valid most of the time, we use it as an internal cache when tracing. `f_back` can be `NULL` even if the Python attribute is not `None`. Setting `f_trace_lines` or `f_trace_opcodes` may not do anything, as the change won't visible to the VM.

Reply:
I tried to clarify that these members are part of the **Python** API.

> Setting f_trace_lines or f_trace_opcodes may not do anything, as the change won't visible to the VM.

Well, it's technically possible to do that in pure Python. My idea is just to explain that the Python API can be accessed in C as well. If there is a bug, it should be either documented or fixed. For example, disallow setting f_trace_lines and f_trace_opcodes.

Currently, f_trace_lines and f_trace_opcodes are not documented. IMO it's better to document them. If there are limits, we can document them.

Reply:
The struct fields cannot be *safely* accessed by C code. The limit is that you can't do it.
The Python attributes can be safely modified by `PyObject_GetAttr` etc.